smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/ExclusiveStructureMemberTraitValidator.java

```diff
@@ -51,30 +51,30 @@
         }
 
         List<ValidationEvent> events = new ArrayList<>();
-        for (StructureShape shape : model.getStructureShapes()) {
+        model.shapes().filter(shape -> shape instanceof NamedMembers).forEach(shape -> {
```

Comment:
nit: Can this just use a for loop with an if statement? It removes all the stream pipelining abstractions